src/application/providers/NotificationsProvider.jsx

```diff
@@ -0,0 +1,22 @@
+import { useMemo, useState } from "react";
+import { createContext } from "react";
+
+export const NotificationsContext = createContext();
+
+export const NotificationProvider = ({ children }) => {
+  const [notification, setNotification] = useState("");
+
+  const value = useMemo(
+    () => ({
+      notification,
+      setNotification,
+    }),
+    [notification, setNotification]
+  );
+
+  return (
+    <NotificationsContext.Provider value={value}>
+      {children}
+    </NotificationsContext.Provider>
+  );
+};
```

Comment:
in an real world scenario, it would be better if you kept the notification stuff (providers and hooks) in a separate PR. 
no need to change it now, just a note to remember the next times ;)